### PR TITLE
Egress gateway performance blog

### PR DESCRIPTION
**Pull request by jnativio**

### CODE REVIEW

**Comment by geeknoid** [comment]
Deploy preview for *preliminary-istio* failed.

Built with commit c7a57986583845b8d2dbcacbb8a6a198182ef55d

https://app.netlify.com/sites/preliminary-istio/deploys/5c126bacf00f9a00078d4c0a

File: content/blog/2018/egress-performance/index.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+{{< image width="75%" ratio="60%"
+    link="./cpu_usage3.png"
+    caption="CPU usage normalized by TPS"
+    >}}
+
+In terms of CPU consumption per transaction, Istio has used significant more CPU only in the egress-gw + SNI proxy case.
+
+## Conclusion
+
+In this investigation, we tried different options to access an external TLS-enabled MongoDB to compare their performance. The introduction of the Ingress Gateway did not cause a significant impact to the performance nor meaningful additional CPU consumption. Only when enabling mTLS between sidecars and egress gateway or using an additional SNI proxy for wildcarded domains we could observe some degradation.
```
Comment (supereagle) [inline content/blog/2018/egress-performance/index.md]: `Ingress Gateway` -> `Egress Gatewa`?

File: .spelling [hunk 2]
```diff
@@ -149,6 +150,7 @@ Reviewer2
 SNI
 SREs
 Secura
+ServiceEntry
```
Comment (geeknoid) [inline .spelling]: Don't want in the dictionary. In the text, either put backticks around ServiceEntry, or simply expand to 'service entry' as appropriate.

File: content/blog/2018/egress-performance/index.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
+---
+title: Egress Gateway Performance Investigation
+description: Verifies the performance impact of adding an egress-gateway.
```
Comment (geeknoid) [inline content/blog/2018/egress-performance/index.md]: egress-gateway -> egress gateway

File: content/blog/2018/egress-performance/index.md [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+{{< image width="70%" ratio="60%"
+    link="./acmeair_regpatrol3.png"
+    caption="Acmeair benchmark in the Istio performance regression patrol environment"
+    >}}
+
+Another difference is that the application communicates with the external DB with plain MongoDB protocol. The first change made for this study was to establish a TLS communication between the MongoDB and its clients running within the application, as this is a more realistic scenario.
+
+Several cases for accessing the external database from the mesh were tested and described next.
+
+## Egress Traffic Cases
```
Comment (geeknoid) [inline content/blog/2018/egress-performance/index.md]: Here and elsewhere, only capitalize the first letter of headers (unless using a proper noun or acronym)

File: content/blog/2018/egress-performance/index.md [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+
+### Case 1:  By-passing the sidecar
```
Comment (geeknoid) [inline content/blog/2018/egress-performance/index.md]: Bypassing, no hyphen

**Comment by geeknoid** [comment]
- There's a typo to fix per the CI failure.

- Please rebase to fix the other CI failures.

- Please move this into the 2019 directory and update the publication date.

- For any images, you no longer need to include the ratio=XX values in there, those are computed automatically unless the image is coming from the Internet.

File: content/blog/2018/egress-performance/index.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+### Throughput
+
+The chart below shows the throughput obtained for the different cases:
+
+{{< image width="75%" ratio="60%"
+    link="./throughput3.png"
+    caption="Throughput obtained for the different cases"
+    >}}
+
+As we can see, there is no major impact in having sidecars and the egress gateway between the application and the external MongoDB, but enabling mTLS and then adding the SNI proxy caused a degradation in the throughput of about 10% and 24%, respectively.
```
Comment (geeknoid) [inline content/blog/2018/egress-performance/index.md]: we -> you

Comment (geeknoid) [reply content/blog/2018/egress-performance/index.md]: mTLS -> mutual TLS
(here and anywhere else)

Comment (jnativio) [reply content/blog/2018/egress-performance/index.md]: ok, fixed the typo and the new comments.
how do I rebase? 
Can it stay in the 2018 folder? It was supposed to be published in December, butdue vacations and other issues, I could not go back to it. The build it was based on is kind of old now.
I don't need use ratio for images, but is it okay to keep it now?

**Comment by geeknoid** [comment]
I rebase with:

   $ git fetch
   $ git rebase

It would be preferable to remove the ratios at that makes this consistent with the rest of the site content and avoids surprises when someone decides to update the image.

As for the date, I think this needs to be the approximate date of publication and not the original date of authoring. It's OK to put a remark in there stating that the number were collected from a particular build in December and that numbers may be different on a more recent build.

Thanks.

**Pull request merged**